projects/plugins/jetpack/extensions/plugins/sharing/sharing.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Block Editor - Sharing feature.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Sharing;
+
+use Jetpack_Gutenberg;
+
+/**
+ * Register Sharing plugin.
+ *
+ * @return void
+ */
+function register_plugins() {
+	// Register Sharing.
+	Jetpack_Gutenberg::set_extension_available( 'sharing' );
+}
+
+add_action( 'jetpack_register_gutenberg_extensions', __NAMESPACE__ . '\register_plugins' );
+
+/**
+ * Allow display the Sharing UI in the editor.
```

Comment:
Maybe it would be worth extending that comment a bit to explain why we need this in relation with `PostTypeSupportCheck`?

```suggestion
 * The Sharing panel is only displayed for post types that support sharing.
 * The sharing module declares support for sharing for all the public post types.
 * Let's do the same thing when the module isn't active yet.
```